portfolio/src/main/java/com/google/sps/comments/NewCommentServlet.java

```diff
@@ -61,8 +64,21 @@ private void addNewComment(HttpServletRequest request) {
     commentEntity.setProperty("message", commentMessage);
     commentEntity.setProperty("votes", 0);
 
+    // Determine the sentiment score of the comment
+    double sentimentScore = sentimentAnalyzer(commentMessage);
+
+    commentEntity.setProperty("sentimentScore", sentimentScore);
+
     // Put entity into datastore.
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(commentEntity);
   }
+
+  // Analyze the overall sentiment of the comment message
+  private double sentimentAnalyzer(String commentMessage) throws IOException {
+    try (LanguageServiceClient languageService = LanguageServiceClient.create()) {
+      Document document = Document.newBuilder().setContent(commentMessage).setType(Document.Type.PLAIN_TEXT).build();
```

Comment:
nit: common convention for a builder pattern here is to put each "set" on a new line and align the periods. This makes it easier to read especially when there are many things to set. This case isn't too bad because it's only a few lines.

```Java
Document document = Document.newBuilder()
    .setContent(commentMessage)
    .setType(Document.Type.PLAIN_TEXT)
    .build();
```

Reply:
Got it, Thank you for the recommendation